rasterio/_io.pyx

```diff
@@ -899,23 +898,20 @@ cdef class MemoryFileBase(object):
         cdef bytes result
 
         try:
-            with CPLErrors() as cple:
-                fp = VSIFOpenL(self.path, 'r')
-                cple.check()
+            fp = VSIFOpenL(self.path, 'r')
 
-                if fp == NULL:
-                    raise IOError(
-                        "Failed to open in-memory file: %s", self.name)
+            if fp == NULL:
```

Comment:
Does `VSIFOpenL()` not trigger a GDAL error that could be grabbed with `exc_wrap_pointer()`